flexible_event_config.md

```diff
@@ -0,0 +1,303 @@
+
+The default configuration for event and navigation sources may not be ideal for all use-cases. We can optionally support extended configuration that allows for callers to specify precisely the information they want out of reports, in order to more efficiently extract utility out of the privacy mechanism. The most efficient configuration will differ from use-case to use-case and will depend on a) the parameters of our privacy mechanism and b) the noise level that can be tolerated by the use-case. For instance, this proposal could be used to:
+
+* Vary the frequency of reports by specifying the number of reporting windows — See e.g. [Binary with more frequent reporting](#binary-with-more-frequent-reporting)
+* Vary the trigger data cardinality in a report — See e.g. a config which matches a [default event source](#default-event-sources)
+* Vary the number of attributions per source registration — See e.g. [Reporting trigger counts](#reporting-trigger-counts) 
+* Reduce the amount of total noise by decreasing the above parameters
+* Configure reporting windows rather than using the defaults — See e.g. [Binary with more frequent reporting](#binary-with-more-frequent-reporting)
+
+
+## Goals
+
+In general, the approach here is to more flexibly encode the output of the API for event-level reports:
+
+* Allow flexibly tuning the output space for a particular source, trading off richness of output (e.g. cardinality of `trigger_data`, number of reporting windows, etc.) with noise levels (See issue #99, issue #734, and issue #733)
+* Allow for value-based reporting (via bucketization) in addition to count-based reporting (issue #55)
+* Allow for choosing flexible reporting windows (issue #46 and issue #736)
+* Allow for setting the possible output space for a source based on a _prior probability distribution_ the ad tech has about the types of trigger that are possible, taking advantage of recent research into label differential privacy for [binary classification](https://arxiv.org/abs/2102.06062) and [regression problems](https://arxiv.org/abs/2212.06074).
+
+
+## API changes 
+
+We will add another optional parameter `trigger_specs` to the JSON in `Attribution-Reporting-Register-Source`:
+
+```jsonc
+{
+  // A trigger spec is a set of matching criteria, along with a scheme to
+  // generate bucketized output based on accumulated values across multiple
+  // triggers within the specified event_report_window.
+  // There will be a limit on the number of specs possible to define for a source.
+  "trigger_specs": [{
+    // This spec will only apply to registrations that set one of the given
+    // trigger data values (positive integers) in the list.
+    // trigger_data will still appear in the event-level report.
+    "trigger_data": [<int>, ...]
+
+    // Represents a series of time windows, starting at <start time>.
+    // At the end of each window, a report for this spec will be delivered an
```

Comment:
Do we want to rephrase this as: a report for this spec will be delivered an hour after the end of each window OR clarify: At the end of each window a report for this spec will be generated that will be delivered an hour later?

Reply:
Ah great call out. This is actually really badly worded. I used your first suggestion with a minor clarification that multiple reports can be generated for each window.